server/portal/apps/workspace/api/views.py

```diff
@@ -48,37 +48,38 @@ def _app_license_type(app_id):
 
 def _get_app(app_id, user):
     agave = user.agave_oauth.client
-    data = agave.apps.get(appId=app_id)
+    data = {}
+    data['definition'] = agave.apps.get(appId=app_id)
 
     # GET EXECUTION SYSTEM INFO FOR USER APPS
-    exec_sys = ExecutionSystem(agave, data['executionSystem'])
+    exec_sys = ExecutionSystem(agave, data['definition']['executionSystem'])
     data['resource'] = exec_sys.login.host
     data['scheduler'] = exec_sys.scheduler
     data['exec_sys'] = exec_sys.to_dict()
 
     # set maxNodes from system queue for app
-    if (data['parallelism'] == 'PARALLEL') and ('defaultQueue' in data):
+    if (data['definition']['parallelism'] == 'PARALLEL') and ('defaultQueue' in data):
         for queue in exec_sys.queues.all():
             if queue.name == data['defaultQueue']:
-                data['maxNodes'] = queue.maxNodes
+                data['definition']['maxNodes'] = queue.maxNodes
                 break
 
     lic_type = _app_license_type(app_id)
-    data['license'] = {
+    data['definition']['license'] = {
         'type': lic_type
     }
     if lic_type is not None:
         _, license_models = get_license_info()
         license_model = list(filter(lambda x: x.license_type == lic_type, license_models))[0]
         lic = license_model.objects.filter(user=user).first()
-        data['license']['enabled'] = lic is not None
+        data['definition']['license']['enabled'] = lic is not None
 
     # Update any App Tray entries upon app retrieval, if their revision numbers have changed
-    matching = AppTrayEntry.objects.all().filter(name=data['name'])
+    matching = AppTrayEntry.objects.all().filter(name=data['definition']['name'])
     if len(matching) > 0:
         first_match = matching[0]
-        if first_match.lastRetrieved and first_match.lastRetrieved != data['id']:
-            data['lastRetrieved'] = first_match.lastRetrieved
+        if first_match.lastRetrieved and first_match.lastRetrieved != data['definition']['id']:
+            data['definition']['lastRetrieved'] = first_match.lastRetrieved
```

Comment:
should `lastRetreived` be outside of definition?

Reply:
I'm not 100% sure, but I think @rstijerina might know. It looks like this `lastRetrieved` key is used to store the latest version of a published app for the apps tray. I couldn't really track down where this was being digested on the front end.

Reply:
Yes, `lastRetrieved` can be removed from the definition - it's not a part of the tapis definition, but we'll still need it in the response for getting